pangea-sdk/v3/service/authn/api.go

```diff
@@ -0,0 +1,1419 @@
+	Type      string      `json:"type"`
+	Life      int         `json:"life"`
+	Expire    string      `json:"expire"`
+	Identity  string      `json:"identity"`
+	Email     string      `json:"email"`
+	Scopes    Scopes      `json:"scopes"`
+	Profile   ProfileData `json:"profile"`
+	CreatedAt string      `json:"created_at"`
+}
+
+// @summary Check a token
+//
+// @description Look up a token and return its contents.
+//
+// @operationId authn_post_v1_client_token_check
+//
+// @example
+//
+//	input := authn.ClientTokenCheckRequest{
+//		Token: "ptu_wuk7tvtpswyjtlsx52b7yyi2l7zotv4a",
+//	}
+//
+//	resp, err := authcli.Client.Token.Check(ctx, input)
+func (a *ClientToken) Check(ctx context.Context, input ClientTokenCheckRequest) (*pangea.PangeaResponse[ClientTokenCheckResult], error) {
+	return request.DoPost(ctx, a.Client, "v2/client/token/check", &input, &ClientTokenCheckResult{})
+}
+
+type ClientPasswordChangeRequest struct {
+	// Base request has ConfigID for multi-config projects
+	pangea.BaseRequest
+
+	Token       string `json:"token"`
+	OldPassword string `json:"old_password"`
+	NewPassword string `json:"new_password"`
+}
+
+type ClientPasswordChangeResult struct {
+}
+
+// @summary Change a user's password
+//
+// @description Change a user's password given the current password.
+//
+// @operationId authn_post_v1_client_password_change
+//
+// @example
+//
+//	input := authn.ClientPasswordChangeRequest{
+//		Token: "ptu_wuk7tvtpswyjtlsx52b7yyi2l7zotv4a",
+//		OldPassword: "hunter2",
+//		NewPassword: "My2n+Password",
+//	}
+//
+//	resp, err := authncli.Client.Password.Change(ctx, input)
+func (a *ClientPassword) Change(ctx context.Context, input ClientPasswordChangeRequest) (*pangea.PangeaResponse[ClientPasswordChangeResult], error) {
+	return request.DoPost(ctx, a.Client, "v2/client/password/change", &input, &ClientPasswordChangeResult{})
+}
+
+type IDProvider string
+
+const (
+	IDPFacebook        IDProvider = "facebook"
+	IDPGithub                     = "github"
+	IDPGoogle                     = "google"
+	IDPMicrosoftOnline            = "microsoftonline"
+	IDPPassword                   = "password"
+)
+
+type MFAProvider string
+
+const (
+	MFAPTOTP     MFAProvider = "totp"
+	MFAPEmailOTP             = "email_otp"
+	IDPSMSOTP                = "sms_otp"
+)
+
+type FlowType string
+
+const (
+	FTsignin FlowType = "signin"
+	FTsignup          = "signup"
+)
+
+type ProfileData map[string]string
+type Scopes []string
+
+type UserCreateRequest struct {
+	// Base request has ConfigID for multi-config projects
+	pangea.BaseRequest
+
+	Email         string       `json:"email"`
+	Authenticator string       `json:"authenticator"`
+	IDProvider    IDProvider   `json:"id_provider"`
+	Verified      *bool        `json:"verified,omitempty"`
+	RequireMFA    *bool        `json:"require_mfa,omitempty"`
+	Profile       *ProfileData `json:"profile,omitempty"`
+	Scopes        *Scopes      `json:"scopes,omitempty"`
+}
+
+type UserCreateResult struct {
```

Comment:
The `UserCreateResult` struct seems like it also needs to be updated. Currently it is:

```go
type UserCreateResult struct {
	ID           string        `json:"id"`
	Email        string        `json:"email"`
	Profile      ProfileData   `json:"profile"`
	IDProviders  []string      `json:"id_providers"`
	RequireMFA   bool          `json:"require_mfa"`
	Verified     bool          `json:"verified"`
	LastLoginAt  *string       `json:"last_login_at,omitempty"`
	Disabled     bool          `json:"disabled"`
	MFAProviders []MFAProvider `json:"mfa_providers,omitempty"`
	CreatedAt    string        `json:"created_at"`
}
```

But looking at the API docs gives:

![Screenshot 2023-10-19 at 3 45 58 PM](https://github.com/pangeacyber/pangea-go/assets/10459856/7f864847-179b-4aba-afa9-8303ced35b2e)

Reply:
Fixed!